src/gwc/src/main/java/org/geoserver/gwc/controller/GwcWmtsRestUrlHandlerMapping.java

```diff
@@ -36,7 +41,7 @@ protected void registerHandlerMethod(
             Object handler, Method method, RequestMappingInfo mapping) {
         // this handler is only interested on GWC WMTS REST API URLs
         for (String pattern : mapping.getPatternsCondition().getPatterns()) {
-            if (pattern.contains("/gwc/rest/wmts")) {
+            if (pattern.contains(handlerMappingString)) {
                 // this is an handler for GWC WMTS REST API
```

Comment:
Suggest constant GWC_WMTS_QUERY, but this code is hard to read with a variable called pattern (not your fault just a comment on the code).

Reply:
changed to GWC_URL_PATTERN